### PR TITLE
Fix bug with sliced input to softmax

### LINKED ISSUE #657: [BUG] Softmax on slice giving wrong result
```python
import mlx.core as mx

s = 4

gates = mx.random.uniform(shape=(s, 4))
scores = mx.softmax(gates[..., 0:2], axis=-1)
print(scores.sum())

inds = mx.broadcast_to(mx.array([0, 1]), (s, 2))
s2 = mx.take_along_axis(gates, inds, axis=-1)
scores = mx.softmax(s2, axis=-1)
print(scores.sum())
```

Tried it on 2 machines (m1 max, m2 ultra) with mlx 0.2.0, both produce the following:
```
array(2, dtype=float32)
array(4, dtype=float32)
```

The second version is right the result should be `4`.

### PR DESCRIPTION
**Pull request by awni**
## Proposed changes

Closes #657

### CODE REVIEW

**Review by angeloskath — APPROVED** [review]
Nice thanks!

File: mlx/backend/accelerate/softmax.cpp [hunk 1]
```diff
@@ -274,7 +274,12 @@ void Softmax::eval_cpu(const std::vector<array>& inputs, array& out) {
 
   // Make sure that the last dimension is contiguous
   auto check_input = [](array x) {
-    if (x.strides()[x.ndim() - 1] == 1) {
+    bool no_copy = x.strides()[x.ndim() - 1] == 1;
+    if (x.ndim() > 1) {
+      auto s = x.strides()[x.ndim() - 1];
```
Comment (angeloskath) [inline mlx/backend/accelerate/softmax.cpp:279]: I think this should be `x.ndim() - 2` btw.

Comment (awni) [reply mlx/backend/accelerate/softmax.cpp:279]: Ah shoot.. you're absolutely right

Comment (awni) [reply mlx/backend/accelerate/softmax.cpp:279]: Let me send another fix...

**Pull request merged**